### PR TITLE
[PWD-2491] Register Program Dialog

### PR DESCRIPTION
**Pull request by gangsararyop**
### What type of PR is this? (_check all applicable_)

- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update
- [ ] Setup
- [ ] Code Refactor
- [ ] Performance Improvements
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Revert

### Screenshots (_optional_)

image

image

### Notes

Nitip jugaa buat api nya gess
https://github.com/PurwadhikaDev/purwadhika-api-new/pull/1081

### CODE REVIEW

File: src/components/PWDKPhoneNumber.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  lineHeight: string
+}
+
+interface ReactPhoneInputProps extends ComponentProps<typeof PhoneInput> {
+  sizeCustom: SizeCustom
+  error: boolean
+}
+
+type PWDKPhoneNumberProps = {
+  size?: string
```
Comment (shadiqfauzi) [inline src/components/PWDKPhoneNumber.tsx:24]:
```suggestion
  size?: string
```
ini bisa dikasih pilihan kan ya sizenya apa aja yg avaiable

File: src/components/RegisterProgram/formik-hooks/index.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { FormikProps, useFormik } from 'formik'
+import { Dispatch, SetStateAction } from 'react'
+import * as Yup from 'yup'
+import { phoneRegExp } from '~/components/PWDKPhoneNumber'
+
+export type Values = {
```
Comment (shadiqfauzi) [inline src/components/RegisterProgram/formik-hooks/index.ts:6]: values disini terlalu generic, bisa di specify ini value apa?

File: src/features/program/api/useGetProgram.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    ProgramResponse,
+    (number | string)[]
+  >
+) => {
+  const { programId } = params
+
+  const { axios, api } = useApiClient()
+
+  return useQuery(
+    ['online-course', programId],
```
Comment (shadiqfauzi) [inline src/features/program/api/useGetProgram.ts:19]:
```suggestion
    ['online-course', programId],
```
key nya online-course?

Comment (gangsararyop) [reply src/features/program/api/useGetProgram.ts:19]: astaga okeehh, efek copas nih gw format nya wkwk

File: src/components/PWDKStepper.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+}))
+
+// ----------------------- components ------------------------
+
+const StepIcon: FC<StepIconProps> = ({ active, completed, icon }) => {
+  return (
+    <RoundedStep
+      ownerState={{ completed, active }}
+      sx={{
+        width: { xs: '24px', md: '32px' },
```
Comment (Vesals) [inline src/components/PWDKStepper.tsx:73]: yo, web CT kan ada guard biar gbs diakses below md, jadi apa breakpoints xs ini akan kepake?

Comment (gangsararyop) [reply src/components/PWDKStepper.tsx:73]: okeeh, lupa nge-take out gw ambil dr v2

**Review by shadiqfauzi — APPROVED** [review]

**Review by kuntiarso — APPROVED** [review]

**Pull request merged**